scripts/mockGuardianSpy.ts

```diff
@@ -0,0 +1,347 @@
+// eslint-disable-next-line import/no-unresolved
+const wormholeBridgeForColonyAbi = require("../artifacts/contracts/bridging/WormholeBridgeForColony.sol/WormholeBridgeForColony.json").abi;
+
+function ethereumAddressToWormholeAddress(address: string) {
+  return ethers.utils.hexZeroPad(ethers.utils.hexStripZeros(ethers.utils.hexlify(address)), 32);
+}
+
+class MockBridgeMonitor {
+
+    homeRpc: string;
+    foreignRpc: string;
+    homeBridgeAddress: string;
+    foreignBridgeAddress: string;
+    homeColonyBridgeAddress: string;
+    foreignColonyBridgeAddress: string;
+    homeBridge: any;
+    foreignBridge: any;
+    homeWormholeBridgeForColony: any;
+    foreignWormholeBridgeForColony: any;
+    skipCount: number = 0;
+    queue: any[] = [];
+    skipped: any[] = [];
+    locked: boolean = false;
+    bridgingPromiseCount: number= 0;
+    resolveBridgingPromise: any;
+    signerHome: any;
+    signerForeign: any;
+    server: Server;
+    subscription: any;
+    subscriptionFilters: FilterEntry[] = [];
+    relayerAddress: string;
+
+  /**
+   * Constructor for (local) bridge monitor. Monitors specified bridge contracts that bridges
+   * messages across in both directions. Does no validation or signing, so only to be used for
+   * local development.
+   * @param {string} homeRpc              The endpoint that the home chain can be queried on
+   * @param {string} foreignRpc           The endpoint that the foreign chain can be queried on
+   * @param {string} homeBridgeAddress    The address of the home bridge contract
+   * @param {string} foreignBridgeAddress The address of the foreign bridge contract
+   * @param {string} homeColonyBridgeAddress The address of the home colony bridge contract
+   * @param {string} foreignColonyBridgeAddress The address of the foreign colony bridge contract
+   */
+  constructor(homeRpc: string, foreignRpc: string, homeBridgeAddress: string, foreignBridgeAddress: string, homeColonyBridgeAddress: string, foreignColonyBridgeAddress: string) {
+    this.homeRpc = homeRpc;
+    this.foreignRpc = foreignRpc;
+    this.homeBridgeAddress = homeBridgeAddress;
+    this.foreignBridgeAddress = foreignBridgeAddress;
+    this.homeColonyBridgeAddress = homeColonyBridgeAddress;
+    this.foreignColonyBridgeAddress = foreignColonyBridgeAddress;
+
+    this.setupListeners();
+
+    this.server = new Server();
+
+    this.subscription;
+
+    this.server.addService(SpyRPCServiceService, {
+        subscribeSignedVAA: (subscription: any, x:any) => {
+          this.subscription = subscription;
+          this.subscriptionFilters = subscription.request.filters;
+          // setTimeout(() => {
+          //     subscription.write({vaaBytes: Buffer.from(vaa, 'base64')});
+          // }, 1000);
+        }
+    });
+
+    this.relayerAddress = "0x770656997AE1C8AB0250571ca1dff5c5A3C37700";
+
+
+    this.server.bindAsync('0.0.0.0:7073', ServerCredentials.createInsecure(), () => {
```

Comment:
Would it make sense to define the wormhole port as a constant and import it throughout?

Reply:
I guess we could... it's only used in three places though, and this is the port it's expected to be on (from the examples / docs).